### PR TITLE
FEATURE: email notifications for policies

### PR DESCRIPTION
**Pull request by heymarkreeves**
- Adds user preference for policy email notifications (always|when_away|never)
- Follows the reminder/notification schedule for sending those

### CODE REVIEW

File: lib/email_controller_helper/policy_email_unsubscriber.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module EmailControllerHelper
+  class PolicyEmailUnsubscriber < BaseEmailUnsubscriber
```
Comment (romanrizzi) [inline lib/email_controller_helper/policy_email_unsubscriber.rb:4]: Nice! Glad to see this used on another plugin.

File: spec/mailers/policy_email_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+describe Jobs::UserEmail do
+  fab!(:user1) { Fabricate(:user) }
+
+  fab!(:group1) do
+    group = Fabricate(:group)
+    group.add(user1)
+    group
+  end
+
+  it "sends a policy alert email to users who have opted in" do
```
Comment (romanrizzi) [inline spec/mailers/policy_email_spec.rb:14]: Can we also test the scenario where the user already accepted the policy?

File: spec/models/post_policy_spec.rb [hunk 3]
```diff
@@ -108,4 +108,58 @@
       expect(policy.not_accepted_by).to_not include(inactive_user, suspended_user)
     end
   end
+
+  describe "#emailed_by" do
```
Comment (romanrizzi) [inline spec/models/post_policy_spec.rb:112]: We test different methods here, so we should use a `describe` for each one.

**Review by romanrizzi — COMMENTED** [review]
PR looks great! I left some minor comments and will approve once we address them

File: spec/mailers/policy_email_spec.rb (hunk 2 above)
Comment (romanrizzi) [reply spec/mailers/policy_email_spec.rb:14]: Oh, this is tested on the `post_policy_spec`. Ignore the comment.

File: spec/models/post_policy_spec.rb (hunk 3 above)
Comment (heymarkreeves) [reply spec/models/post_policy_spec.rb:112]: This has been updated. I think it was the only change requested? Thanks again!

**Review by romanrizzi — APPROVED** [review]

**Pull request merged**